src/lib/operators/table_scan.cpp

```diff
@@ -94,11 +94,12 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
       const auto chunk_guard = _in_table->get_chunk_with_access_counting(chunk_id);
       // The actual scan happens in the sub classes of BaseTableScanImpl
       const auto matches_out = std::make_shared<PosList>(_impl->scan_chunk(chunk_id));
+      if (matches_out->empty()) return;
 
-      // The output chunk is allocated on the same NUMA node as the input chunk. Also, the AccessCounter is
+      // The output chunk is allocated on the same NUMA node as the input chunk. Also, the ChunkAccessCounter is
       // reused to track accesses of the output chunk. Accesses of derived chunks are counted towards the
       // original chunk.
-      auto chunk_out = std::make_shared<Chunk>(chunk_guard->get_allocator(), chunk_guard->access_counter());
+      ChunkColumns out_columns;
```

Comment:
How do you ensure that what is written above is still true? What did this guard do anyway?

Reply:
The "The output chunk is allocated on the same NUMA node as the input chunk" bit is a lie as far as I can tell, look at the place where the pos list is allocated. Removed that part of the comment. I will open an issue requesting the retirement of the current NUMA-awareness, but that's not part of this PR